plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/handlers/FordiacQuickFixHandler.java

```diff
@@ -91,10 +107,34 @@ public Object execute(final ExecutionEvent event) {
 			QuickFixWizardDialog.openDialog(shell, selectedMarkers, resolutions, resolutionsMap);
 		}
 		deleteTemporaryMarker(selectedMarkers);
-
 		return null;
 	}
 
+	private static IMarker[] getSelectedVdMarkersFromNatTable(final IStructuredSelection selection) {
+		if (selection.getFirstElement() instanceof final EObject obj) {
+			return retrieveMarkerFromObject(obj);
+		}
+		return new IMarker[0];
+	}
+
+	public static IMarker[] retrieveMarkerFromObject(final Object obj) {
+		final IEditorPart editor = EditorUtils.getCurrentActiveEditor();
+		if (null != editor) {
+			final GraphicalViewer gfv = editor.getAdapter(GraphicalViewer.class);
```

Comment:
You don't need the viewer you don't need the edit part. Just use:
`GraphicalAnnotationModel annotationModel = activEditor.getAdpater(GraphicalAnnotationModel.class);`
this is exactly what is happeing in FordiacAnnotationUtil